packages/material-ui/src/utils/useControlled.test.js

```diff
@@ -83,4 +83,11 @@ describe('useControlled', () => {
       'Material-UI: A component is changing the default value state of an uncontrolled TestComponent after being initialized.',
     );
   });
+
+  it('should not raise a warning if changing the defaultValue when controlled', () => {
+    const { setProps } = render(<TestComponent value={[]} defaultValue={[]}>{() => null}</TestComponent>);
+    expect(consoleErrorMock.callCount()).to.equal(0);
+    setProps({ value: null, defaultValue: null });
```

Comment:
I haven't tested this but this should work in theory.

```suggestion
    const { setProps } = render(<TestComponent value={true} defaultValue={false}>{() => null}</TestComponent>);
    expect(consoleErrorMock.callCount()).to.equal(0);
    setProps({ defaultValue: true });
```

But that being said wouldn't the new `defaultValue` be ignored anyway? Maybe the warning should be amended not to talk about `uncontrolled`?

Reply:
Thanks for assistance